### PR TITLE
Add a brief doc and sample code describing how a watcher can re-connect to etcd server after disconnected

### PR DESCRIPTION
**Pull request by niyue**
Add some brief doc to answer the question in https://github.com/etcd-cpp-apiv3/etcd-cpp-apiv3/issues/73

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -487,7 +487,47 @@ either by user implicitly calling ```Cancel()``` or when watcher class is destro
   watcher.Cancel();
   etcd.set("/test/key", "43"); /* print response will NOT be called,
                                   since watch is already cancelled */
```
Comment (niyue) [inline README.md:489]: A unnecessary curly bracket in the `Watcher Class` section's code sample is removed

**Comment by sighingnow** [comment]
Thank you @niyue !

**Comment by sighingnow** [comment]
See also revision in #124.

**Comment by sighingnow** [comment]
See also revision in #124.

**Pull request merged**